// @yuheiy/responsive-image-map/lib/utils.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.debounce = exports.detectImage = undefined;

var _promise = require('babel-runtime/core-js/promise');

var _promise2 = _interopRequireDefault(_promise);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var detectImage = exports.detectImage = function detectImage(src) {
  return new _promise2.default(function (resolve, reject) {
    var tester = new Image();
    var onLoad = function onLoad() {
      removeListeners();
      resolve();
    };
    var onError = function onError() {
      removeListeners();
      reject();
    };
    var removeListeners = function removeListeners() {
      tester.removeEventListener('load', onLoad);
      tester.removeEventListener('error', onError);
    };

    tester.addEventListener('load', onLoad);
    tester.addEventListener('error', onError);
    tester.src = src;
  });
};

var debounce = exports.debounce = function debounce(fn) {
  var scheduledAnimationFrame = false;

  return function () {
    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    if (scheduledAnimationFrame) {
      return;
    }

    scheduledAnimationFrame = true;
    requestAnimationFrame(function () {
      fn.apply(undefined, args);
      scheduledAnimationFrame = false;
    });
  };
};
